thanks-bot/bot.js

```diff
@@ -1,35 +1,59 @@
-var Discord = require('discord.io');
-var logger = require('winston');
-var auth = require('./auth.json');
+const Discord = require('discord.io')
+const logger = require('winston')
+const auth = require('./auth.json')
 
 // Configure logger settings
-logger.remove(logger.transports.Console);
+logger.remove(logger.transports.Console)
 logger.add(logger.transports.Console, {
-    colorize: true
-});
+  colorize: true
+})
 
-logger.level = 'debug';
+logger.level = 'debug'
 
 // Initialize Discord Bot
-var bot = new Discord.Client({
-   token: auth.token,
-   autorun: true,
-});
+const bot = new Discord.Client({
+  token: auth.token,
+  autorun: true
+})
 
 bot.on('ready', function (evt) {
-    logger.info('Connected');
-    logger.info('Logged in as: ');
-    logger.info(bot.username + ' - (' + bot.id + ')');
-});
+  logger.info('Connected')
+  logger.info('Logged in as: ')
+  logger.info(bot.username + ' - (' + bot.id + ')')
+})
+
+const ALLOWED_CHANNEL_ID = '525249065039036426'
 
 bot.on('message', function (user, userID, channelID, message, evt) {
-    switch(message) {
-        case 'Ассаляму аляйкум':
-            bot.sendMessage({
-                to: channelID,
-                message: 'Уа аляйкум салям',
-            });
-
-            break;
-    }
-});
+  let thanks_message_regexp = /!thanks\b/gmi
+
+  // Conditions for exiting the functionTHANKS_MESSAGE_REGEXP
+  if (channelID !== ALLOWED_CHANNEL_ID ||
+    userID === bot.id || !thanks_message_regexp.test(message)) {
+    return
+  }
+
+  // Collecting usernames
+  let mentionedUserNames = evt.d.mentions.map(item => item.username)
+
+  // Collecting userIds
+  let mentionedUserIds = evt.d.mentions.map(item => item.id)
```

Comment:
Let's be consistent.
```suggestion
  let mentionedUserIDs = evt.d.mentions.map(item => item.id)
```